test/remote/ups_test.rb

```diff
@@ -218,6 +218,35 @@ def test_obtain_shipping_label_without_dimensions
     assert_instance_of ActiveShipping::LabelResponse, response
   end
 
+  def test_obtain_shipping_label_with_bill_third_party
+    begin
+      bill_third_party_credentials = credentials(:ups_third_party_billing)
+    rescue NoCredentialsFound => e
+      skip(e.message)
+    end
+
+    response = @carrier.create_shipment(
+      location_fixtures[:beverly_hills],
+      location_fixtures[:new_york_with_name],
+      package_fixtures.values_at(:books),
+      {
+        :test => true,
+        :bill_third_party => true,
+        :billing_account => bill_third_party_credentials[:account],
+        :billing_zip => bill_third_party_credentials[:zip],
+        :billing_country => bill_third_party_credentials[:country_code]
+      }
+    )
+
+    assert response.success?
+
+    # All behavior specific to how a LabelResponse behaves in the
+    # context of UPS label data is a matter for unit tests.  If
+    # the data changes substantially, the create_shipment
+    # ought to raise an exception and this test will fail.
+    assert_instance_of ActiveShipping::LabelResponse, response
```

Comment:
When you create_shipment, for the most part, it just doesn't blow up. ShipmentAcceptResponse or ShipmentConfirmResponse do not contain any specific tag indicating that it is a third party billing. There is an indication on the gif label that the billing is "3rd party", but I am not sure how that can be parsed out from GraphicImage and be tested.